src/alchemlyb/estimators/bar_.py

```diff
@@ -97,18 +97,21 @@ def fit(self, u_nk):
             (len(groups.get_group(i)) if i in groups.groups else 0)
             for i in u_nk.columns
         ]
-
+        states = [x for i, x in enumerate(self._states_) if N_k[i] > 0]
         # Now get free energy differences and their uncertainties for each step
         deltas = np.array([])
         d_deltas = np.array([])
         for k in range(len(N_k) - 1):
+            if N_k[k] == 0 or N_k[k + 1] == 0:
```

Comment:
Might be good to have a test for this as well.

Reply:
I'm not sure how to test this directly, so I added an error at the end of the loop so that if this if statement is true for every value of k, a descriptive error is given as to why. This error is tested with `test_block_average_error_4_bar`.